util/headers_periph_gen.py

```diff
@@ -0,0 +1,74 @@
+import structs_gen
+
+# Generates the header file with the structs for every peripheral specified by a json file
+
+template_path = "./sw/device/lib/drivers/template.tpl"     # template used for the output
+json_files = []                                             # list of all the json files
+output_files = []                                           # list of all the output file names
+peripheral_names = []                                       # names of the peripherals for which the structs are generated
+
+# dma
+json_files.append("./hw/ip/dma/data/dma.hjson")
```

Comment:
why instead of hardwiring the hjson files in the script you do not create a HJSON/TXT or whatever file that lists all the HJSON files you need to take as input and pass it to the script? then the script iterates over the HJSON files listed in the array?

Reply:
Davide is right, let's use the mcu_cfg.hjson and add there the fields for every peripheral @StefanoAlbini96